compiler/src/dotty/tools/dotc/typer/TyperPhase.scala

```diff
@@ -103,19 +75,26 @@ class FrontEnd extends Phase {
   override def runOn(units: List[CompilationUnit])(using Context): List[CompilationUnit] =
     val unitContexts =
       for unit <- units yield
-        report.inform(s"compiling ${unit.source}")
-        ctx.fresh.setCompilationUnit(unit).withRootImports
-    unitContexts.foreach(parse(using _))
-    record("parsedTrees", ast.Trees.ntrees)
+        val newCtx = ctx.fresh.setPhase(this.start).setCompilationUnit(unit)
```

Comment:
```suggestion
        report.inform(s"compiling ${unit.source}")
        val newCtx = ctx.fresh.setPhase(this.start).setCompilationUnit(unit)
```

Or perhaps "typing ${unit.source}"?

Reply:
Done